src/overlays/actors/ovl_En_Bom_Chu/z_en_bom_chu.c

```diff
@@ -129,61 +131,72 @@ void EnBomChu_CrossProduct(Vec3f* a, Vec3f* b, Vec3f* dest) {
+    normal.z = COLPOLY_GET_NORMAL(floorPoly->normal.z);
+
+    normDotUp = DOTXYZ(normal, this->axisUp);
+
+    if (!(fabsf(normDotUp) >= 1.0f)) {
+        angle = Math_FAcosF(normDotUp);
 
-    temp_f12 = (sp84.x * this->unk_160.x) + (sp84.y * this->unk_160.y) + (sp84.z * this->unk_160.z);
+        if (!(angle < 0.001f)) {
+            EnBomChu_CrossProduct(&this->axisUp, &normal, &vec);
+            //! @bug this function expects a unit vector but `vec` is not normalized
+            Matrix_RotateAxis(angle, &vec, MTXMODE_NEW);
 
-    if (!(fabsf(temp_f12) >= 1.0f)) {
-        sp74 = Math_FAcosF(temp_f12);
+            Matrix_MultVec3f(&this->axisLeft, &vec);
+            this->axisLeft = vec;
 
-        if (!(sp74 < 0.001f)) {
-            EnBomChu_CrossProduct(&this->unk_160, &sp84, &sp78);
-            Matrix_RotateAxis(sp74, &sp78, 0);
-            Matrix_MultVec3f(&this->unk_16C, &sp78);
-            this->unk_16C = sp78;
-            EnBomChu_CrossProduct(&this->unk_16C, &sp84, &this->unk_154);
+            EnBomChu_CrossProduct(&this->axisLeft, &normal, &this->axisForwards);
 
-            magnitude = Math3D_Vec3fMagnitude(&this->unk_154);
+            magnitude = Math3D_Vec3fMagnitude(&this->axisForwards);
 
             if (magnitude < 0.001f) {
                 EnBomChu_Explode(this, globalCtx);
                 return;
             }
 
-            this->unk_154.x *= (1.0f / magnitude);
-            this->unk_154.y *= (1.0f / magnitude);
-            this->unk_154.z *= (1.0f / magnitude);
+            this->axisForwards.x *= 1.0f / magnitude;
+            this->axisForwards.y *= 1.0f / magnitude;
+            this->axisForwards.z *= 1.0f / magnitude;
 
-            this->unk_160 = sp84;
+            this->axisUp = normal;
 
             if (1) {}
 
-            mf.xx = this->unk_16C.x;
-            mf.yx = this->unk_16C.y;
-            mf.zx = this->unk_16C.z;
+            // mf = (axisLeft | axisUp | axisForwards)
 
-            mf.xy = sp84.x;
-            mf.yy = sp84.y;
-            mf.zy = sp84.z;
+            mf.xx = this->axisLeft.x;
+            mf.yx = this->axisLeft.y;
+            mf.zx = this->axisLeft.z;
 
-            mf.xz = this->unk_154.x;
-            mf.yz = this->unk_154.y;
-            mf.zz = this->unk_154.z;
+            mf.xy = normal.x;
+            mf.yy = normal.y;
+            mf.zy = normal.z;
+
+            mf.xz = this->axisForwards.x;
+            mf.yz = this->axisForwards.y;
+            mf.zz = this->axisForwards.z;
 
             Matrix_MtxFToYXZRotS(&mf, &this->actor.world.rot, 0);
 
+            /*
+             * A hack for preventing bombchus from sticking to ledges.
+             * The visual rotation reverts the sign inversion (shape.rot.x = -world.rot.x).
+             * The better fix would be making func_8002D908 compute XYZ velocity better,
+             * or not using it and make the bombchu compute its own velocity.
+             */
```

Comment:
would personally prefer // comments on each line, and leave that comment styling for higher level function/file descriptions